src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const length = this.length;
+
+    for (let i = 0; i < length; i++) {
+      if (i > 0 && separator !== undefined) {
```

Comment:
You don't need to check if `separator !== undefined`, as you set a default value for `separator`, it will never be equal to `undefined`